### PR TITLE
Add From/TryFrom traits to primitive types

### PR DESCRIPTION
**Pull request by HydrogenMacro**
For #9; added implementation as well as tests.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -668,6 +668,73 @@ impl Hash for i24 {
     }
 }
 
+macro_rules! impl_i24_from_primitive {
+    ($($target_type:ty)*) => {
+	    $(
+		    impl core::convert::From<$target_type> for i24 {
+			    fn from(value: $target_type) -> Self {
+			        i24::from_i32(i32::from(value))
+			            .expect("From is only implemented for signed types with less than 24 bits")
```
Comment (vrtgs) [inline src/lib.rs:677]: this leads to pretty bad code gen especially for bool

File: src/lib.rs [hunk 2]
```diff
@@ -700,10 +715,26 @@ macro_rules! impl_try_from {
...
 }
 
+macro_rules! impl_primitive_try_from_i24 {
+    ($($target_type:ty)*) => {
+	    $(
+		    impl core::convert::TryFrom<i24> for $target_type {
+			    type Error = ParseI24Error;
+			    fn try_from(value: i24) -> Result<Self, Self::Error> {
+				    Self::try_from(value.to_i32()).map_err(|_| ParseI24Error::OutOfRange)
+			    }
```
Comment (vrtgs) [inline src/lib.rs:729]: `ParseI24Error` will be rolled out completely with #8, and again with the bad code gen

File: src/lib.rs [hunk 3]
```diff
@@ -906,51 +937,6 @@ mod i24_tests {
...
-		test!(u32);
-		test!(isize);
-		test!(usize);
-		test!(i64);
-		test!(u64);
-		test!(i128);
-		test!(u128);
-	}
+
+	// TODO: Add tests
```
Comment (vrtgs) [inline src/lib.rs:941]: adding tests is kinda needed

**Review by vrtgs — COMMENTED** [review]
Sadly less than optimal code gen, and no tests

**Review by vrtgs — CHANGES_REQUESTED** [review]
less than optimal code gen, and no tests

File: src/lib.rs (hunk 1 above)
Comment (HydrogenMacro) [reply src/lib.rs:677]: It was meant to be a temporary solution for my use case; I'll optimize it later

Comment (vrtgs) [reply src/lib.rs:677]: dw, fix coming soon been busy lately tho, so took a bit

**Pull request closed without merging**